src/components/RedefinedDomainResolver.tsx

```diff
@@ -0,0 +1,103 @@
+import React, {useCallback, useEffect, useState} from "react";
+import _debounce from 'lodash/debounce';
+import styled from "styled-components";
+import DropDown from "./DropDown";
+import {ContainerProps, RedefinedDomainResolverProps, InputProps} from "../types";
+import companyLogo from '../assets/small-logo.svg';
+import {baseStyle} from "../styles/baseStyle";
+import GlobalStyle from "../styles/globalStyle";
+import {RedefinedResolver} from "redefined-resolver";
+
+const RedefinedDomainResolver = (props: RedefinedDomainResolverProps) => {
+  const {width, height, onSelect} = props;
+  const [dropDownActive, setDropDownActive] = useState(false);
+  const [domain, setDomain] = useState("");
+  const [addresses, setAddresses] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState("");
+
+  useEffect(() => {
+    domain.length === 0 && setDropDownActive(false);
```

Comment:
It is good if we will return sth `return  domain.length === 0 && getShit()`
But without return it is not good code style for ts/js, i think

Reply:
what is `getShift()?`  why is return here? i don`t understand.

Reply:
getShift is an example)
It does not return here, but this entry `return  domain.length === 0 && getShit()` implies a return
Let`s use this if you want to do it in line)
```
if (domain.length === 0) setDropDownActive(false);
```